modules/broker-core/src/main/java/org/wso2/broker/core/Binding.java

```diff
@@ -38,22 +38,22 @@
 
     private final FieldTable arguments;
 
-    private final BooleanExpression jmsFilter;
+    private final BooleanExpression filterExpression;
 
     Binding(Queue queue, String routingKey, FieldTable arguments) throws BrokerException {
         this.queue = queue;
         this.routingKey = routingKey;
         this.arguments = arguments;
         String filterString = null;
         try {
-            MessageFilter messageFilter = new MessageFilter();
             FieldValue fieldValue = arguments.getValue(JMS_SELECTOR_ARGUMENT);
             if (fieldValue != null &&
                     fieldValue.getType() == FieldValue.Type.LONG_STRING &&
                     !(filterString = fieldValue.getValue().toString()).isEmpty()) {
-                jmsFilter = messageFilter.parse(filterString);
+                MessageFilter messageFilter = new MessageFilter(filterString);
+                filterExpression = messageFilter.parse();
             } else {
-                jmsFilter = null;
+                filterExpression = null;
```

Comment:
I think it's better if we can refactor this class in to (filter aware and nonaware) so that we can avoid the null.

Reply:
+1

Reply:
I didn't mean in this PR :wink:

Reply:
Ya. Got that :)